header/header.go

```diff
@@ -174,19 +181,23 @@ func (eh *ExtendedHeader) Verify(untrst *ExtendedHeader) error {
 		// Check the validator hashes are the same
 		if !bytes.Equal(untrst.ValidatorsHash, eh.NextValidatorsHash) {
 			return &libhead.VerifyError{
-				Reason: fmt.Errorf("expected old header next validators (%X) to match those from new header (%X)",
-					eh.NextValidatorsHash,
-					untrst.ValidatorsHash,
-				),
+				Reason: fmt.Errorf("%w: %w",
+					ErrValidatorHashMismatch,
+					fmt.Errorf("expected old header next validators (%X) to match those from new header (%X)",
+						eh.NextValidatorsHash,
+						untrst.ValidatorsHash,
+					)),
```

Comment:
```suggestion
				Reason: fmt.Errorf("%w: expected old header next validators (%X) to match those from new header (%X)",
					ErrValidatorHashMismatch,
					eh.NextValidatorsHash,
					untrst.ValidatorsHash,
					),
```

Reply:
Let's simplify it a bit

Reply:
I would also decrease the verbosity for the error message inside fmt.Errorf and move as much as possible to  ErrValidatorHashMismatch

Reply:
Reduced.